### ISSUE TITLE
Upgrade akka-serialization-jackson to Jackson 2.11.0

### ISSUE THREAD

**Issue by hseeberger** [akka/akka#29072]
See https://github.com/lightbend/cloudstatemachine/issues/979

**Comment by raboof** [comment]
Interesting! We have to make sure that will not introduce any behavior/compatibility changes though - there's some discussion in #28970 but I don't think anyone has really looked closely at the details yet.

**Comment by ennru** [comment]
See https://github.com/akka/akka/pull/28970#issuecomment-671201057

**Comment by henricook** [comment]
Having an old version of Jackson embedded in Akka like this seems to be causing issues for me as a user of Play Framework when I try to use libraries that rely on a newer version of Jackson (e.g. https://github.com/okta/okta-jwt-verifier-java/)

I end up with a runtime error:

```
com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.10.4 requires Jackson Databind version >= 2.10.0 and < 2.11.0
```

**Comment by henricook** [comment]
It's worth noting that several popular libraries (e.g. auth0/jwks-rsa, aws-java-sdk) recently went to 2.11 as well, so now Akka is holding a lot of people back from upgrading

**Comment by raboof** [comment]
> Scala module 2.10.4 requires Jackson Databind version >= 2.10.0 and < 2.11.0

I suspect this might be an error message produced by https://github.com/FasterXML/jackson-module-scala - perhaps upgrading that module as well might help?

Note that we don't promise Akka will work with such newer Jackson versions, but indeed in practice it might work out.

**Comment by pjfanning** [comment]
Would it be possible to consider an upgrade to jackson 2.12.0?
jackson-module-scala 2.12 branch is where development for Scala 3 support is under way and it is unlikely that the changes will be backported. A lot of test code and a reasonable amount of main source has had to be changed to get things to compile. Even with that there are a lot of test failures.

https://github.com/FasterXML/jackson-module-scala/issues/479

I've already deprecated ScalaObjectMapper class because it depends on Manifests and they don't work in Scala 3. So I'd recommend rewriting any akka-serialization-jackson code that relies on ScalaObjectMapper.

**Comment by mpartan** [comment]
> Having an old version of Jackson embedded in Akka like this seems to be causing issues for me as a user of Play Framework 2.8.4 when I try to use libraries that rely on a newer version of Jackson (e.g. https://github.com/okta/okta-jwt-verifier-java/)
> 
> Akka: 2.10.x
> Okta-jwt-verifier version 0.5.0 and above: 2.11.x
> 
> I end up with a runtime error:
> 
> ```
> com.fasterxml.jackson.databind.JsonMappingException: Scala module 2.10.4 requires Jackson Databind version >= 2.10.0 and < 2.11.0
> ```
> 
> Other libraries now using newer versions of Jackson that I've not been able to upgrade because of akka (unless there's some workaround?):
> 
>     * `auth0/jwks-rsa`

Having similar issues when trying to use together with newer Azure Java libraries, which use a more recent version of Jackson.

**Comment by patriknw** [comment]
Have you tried to override *all* jackson modules to 2.11.4, and what runtime error do you see then?

**Comment by pjfanning** [comment]
@mpartan add an explicit dependency on your build script for all the jackson jars that you depend on - they should all be set to the same version

The exception comes from jackson-module-scala - it checks that its version matches that of jackson-databind.

**Comment by henricook** [comment]
I see this MR ^ was closed because "there's another way" - would anyone be kind enough to spell out how this other way might work? With more and more of my dependencies upgrading to 2.11 i'm not holding more than three things back because of the jackson-databind version in akka

**Comment by pjfanning** [comment]
@henricook have you tried explicitly setting the version on all akka dependencies in your build script (see my comment on Jan 13)? I use that approach and akka runs fine for me with jackson 2.12.1.

**Comment by henricook** [comment]
I'm on a scala/sbt/playframework project so I guess I'm looking for some very specific instructions. I forgot for a moment that Akka is, of course, very cross-language! I'll take your advice and try to apply it to my `build.sbt` and post back here if I'm successful.

**Comment by heikobornholdt** [comment]
You are referencing a comment by @dependabot-preview. This is a bot. The comment is misleading and therefore should be ignored.

**Comment by pjfanning** [comment]
can this be closed as it is released in akka 2.6.14?

**Comment by patriknw** [comment]
yes, thanks
